main.nf

```diff
@@ -1,337 +1,52 @@
-#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
 
-dataDir = "${workflow.launchDir}/${params.dataDir}"
 metadataDir = "${workflow.launchDir}/${params.metadataDir}"
 egaCredentialsDir = "${workflow.launchDir}/${params.egaCredentialsDir}"
-fetchMode = params.fetchMode
-
-DATASETS = Channel.fromPath( "$dataDir/EGAD*", type: 'dir', checkIfExists: true )
-
-DATASETS.map{ dir -> tuple( file(dir).getName(), dir ) }
-    .set {KEYED_DATASETS}
 
+workflow {
+    main:
+        fetch_data(params.EGA_DATASET_ID)
+        make_metadata_table(params.EGA_DATASET_ID)
+}
 
-if (fetchMode == 'aspera'){
-    process get_dbox_content {
+process fetch_data {
 
-        cache 'lenient'
-        
-        input:
-            set val(dsId), file(dsPath) from KEYED_DATASETS
+    conda 'pyega3=5.2.0'
 
-        output:
-            set val(dsId), file('dbox_content') into AVAILABLE_FILES
+    cache 'lenient'
 
-        """
-        credentialsFile=$egaCredentialsDir/${dsId}.txt
-        if [ ! -e \$credentialsFile ]; then
-            echo "Credentials file missing at \$credentialsFile" 1>&2
-            exit 1
-        fi
+    input:
+        val EGA_DATASET_ID
 
-        user=\$(grep "^user=" \$credentialsFile | sed s/user=//)
-        password=\$(grep "^password=" \$credentialsFile | sed s/password=//)
-        export ASPERA_SCP_PASS="\$password";   
+    output:
+        file "ega_data/*"
 
-        ascp --ignore-host-key -d -QTl 100m \${user}@xfer.crg.eu:dbox_content \$(pwd)/
+    script:
         """
-    }
-}
-else{
-    process get_ega_file_listing {
-        
-        cache 'lenient'
-
-        conda 'pyega3>=5.0.1'
-
-        input:
-            set val(dsId), file(dsPath) from KEYED_DATASETS
-        
-        output:
-            set val(dsId), file('pyega3_file_listing.txt') into AVAILABLE_FILES  
-
+        mkdir -p ega_data
+        # list unencrypted md5 checksums for all files
+        pyega3 -cf $egaCredentialsDir/ega.credentials files ${params.EGA_DATASET_ID}
```

Comment:
${params.EGA_DATASET_ID} references a parameter from the "params" object not the input value of the process EGA_DATASET_ID. 
Should that be changed by $EGA_DATASET_ID?